client_test.go

```diff
@@ -1,295 +1,333 @@
 	assert.NoError(t, err)
+
+	allFlags := flags.AllFlags()
+
+	assert.Equal(t, 1, len(allFlags))
+
+	assert.Equal(t, fixtures.Feature1Name, allFlags[0].FeatureName)
+	assert.Equal(t, fixtures.Feature1ID, allFlags[0].FeatureID)
+	assert.Equal(t, fixtures.Feature1Value, allFlags[0].Value)
+
 }
 
-func updateTraitsAsserts(updated []*flagsmith.Trait, err error, c *flagsmith.Client, t *testing.T) {
-	assert.NoError(t, err)
-	assert.NotEmpty(t, updated)
-	assert.Equal(t, 5, len(updated))
-	assert.Equal(t, "true", updated[0].Value)
-	assert.Equal(t, "42", updated[1].Value)
-	assert.Equal(t, "foo bar baz", updated[2].Value)
-	assert.Equal(t, "616", updated[3].Value)
-	assert.Equal(t, "3.14", updated[4].Value)
-
-	trait, err := c.GetTrait(differentUser, "boolField")
-	assert.NoError(t, err)
-	assert.Equal(t, "true", trait.Value)
+func TestDefaultHandlerIsUsedWhenNoMatchingEnvironmentFlagReturned(t *testing.T) {
+	// Given
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 
-	trait, err = c.GetTrait(differentUser, "intField")
-	assert.NoError(t, err)
-	assert.Equal(t, "42", trait.Value)
+		assert.Equal(t, req.URL.Path, "/api/v1/flags/")
+		assert.Equal(t, fixtures.EnvironmentAPIKey, req.Header.Get("X-Environment-Key"))
 
-	trait, err = c.GetTrait(differentUser, "stringField")
-	assert.NoError(t, err)
-	assert.Equal(t, "foo bar baz", trait.Value)
+		rw.Header().Set("Content-Type", "application/json")
+
+		rw.WriteHeader(http.StatusOK)
+		_, err := io.WriteString(rw, fixtures.FlagsJson)
+
+		assert.NoError(t, err)
+	}))
+	defer server.Close()
+
+	// When
+	client := flagsmith.NewClient(fixtures.EnvironmentAPIKey, flagsmith.WithBaseURL(server.URL+"/api/v1/"),
+		flagsmith.WithDefaultHandler(func(featureName string) flagsmith.Flag {
+			return flagsmith.Flag{IsDefault: true}
+		}))
 
-	trait, err = c.GetTrait(differentUser, "anotherField")
+	flags, err := client.GetEnvironmentFlags()
+
+	// Then
 	assert.NoError(t, err)
-	assert.Equal(t, "616", trait.Value)
 
-	trait, err = c.GetTrait(differentUser, "floatField")
+	flag, err := flags.GetFlag("feature_that_does_not_exist")
 	assert.NoError(t, err)
-	assert.Equal(t, "3.14", trait.Value)
+	assert.True(t, flag.IsDefault)
 }
 
-func TestBulkUpdateTraitsPointers(t *testing.T) {
-	c := flagsmith.NewClient(apiKey)
-
-	traits := []*flagsmith.Trait{
-		{
-			Identity: flagsmith.User{},
-			Key:      "boolField",
-			Value:    "true",
-		},
-		{
-			Identity: flagsmith.User{},
-			Key:      "intField",
-			Value:    "42",
-		},
-		{
-			Identity: flagsmith.User{},
-			Key:      "stringField",
-			Value:    "foo bar baz",
-		},
-		{
-			Identity: flagsmith.User{},
-			Key:      "anotherField",
-			Value:    "616",
-		},
-		{
-			Identity: flagsmith.User{},
-			Key:      "floatField",
-			Value:    "3.14",
-		},
-	}
-	updated, err := c.UpdateTraits(differentUser, traits)
-	updateTraitsAsserts(updated, err, c, t)
-}
+func TestErrorIsReturnedIfRequestFails(t *testing.T) {
```

Comment:
I guess this is only in the scenario that we don't have a defaultFlagHandler set, right?

Reply:
Yep